lib/kube/proxy/forwarder.go

```diff
@@ -748,6 +819,24 @@ func (f *Forwarder) getKubeAccessDetails(
 	}, nil
 }
 
+// podRegex is the Pods endpoint API url.
+var podRegex = regexp.MustCompile(`(?m)/api/v1/namespaces/([^/]+)/pods/([^/]+)`)
+
+// getPodResourceFromRequest returns a KubernetesResource if the user tried to access
+// a specific Pod endpoint. Otherwise, returns nil.
+// TODO(tigrato): extend it to support other resources.
+func getPodResourceFromRequest(requestURI string) *types.KubernetesResource {
+	if !podRegex.MatchString(requestURI) {
+		return nil
+	}
+	matches := podRegex.FindStringSubmatch(requestURI)
```

Comment:
Avoid matching twice:

```suggestion
	matches := podRegex.FindStringSubmatch(requestURI)
	if matches == nil {
		return nil
	}	
```